doc/development_process/api_lifecycle.rst

```diff
@@ -0,0 +1,129 @@
+.. _api_lifecycle:
+
+API Lifecycle
+#############
+
+Developers using Zephyr's APIs need to know how long they can trust that a
+given API will not change in future releases. At the same time, developers
+maintaining and extending the Zephyr's APIs need to be able to introduce
+new APIs that aren't yet fully proven, and to potentially retire old APIs when they're
+no longer optimal or supported by the underlying platforms.
+
+
+.. figure:: api_lifecycle.png
+    :align: center
+    :alt: API Life Cycle
+    :figclass: align-center
+
+    API Life Cycle
+
+
+Experimental
+*************
+
+Experimental APIs denote that a feature was introduced recently, and may change
+or be removed in future versions. Try it out and provide feedback
+to the community via the `Developer mailing list <https://lists.zephyrproject.org/g/devel>`_.
+
+Peripheral APIs (Hardware Related)
+==================================
+
+When introducing an API (header file with documentation) for a new peripheral
+or driver subsystem, review of the API is enforced and is driven by the API
+working group consisting of representatives from different vendors.
+
+The following requirements apply to new APIs:
+
+- Documentation of the API (Usage)
+  explaining its design and assumptions, how it is to be used, current
+  implementation limitations, and future potential, if appropriate.
+- The API introduction should be accompanied by at least one driver
+  implementing this API
+- At least one sample using the new API
+
+The API shall be promoted to ``unstable`` when it has at least two
+implementations on different hardware platforms.
+
+Hardware Agnostic APIs
+=======================
+
+For hardware agnostic APIs, multiple applications are required to promote an
+API from ``experimental`` to ``unstable``.
+
+
+Unstable
+********
+
+The API is in the process of settling, but has not yet had sufficient real-world
+testing to be considered stable. The API is considered generic in nature and can
+be used on diferrent hardware platforms.
+
+.. note::
+
+   Changes will not be announced.
+
+Stable
+*******
+
+The API has proven satisfactory, but cleanup in the underlying code may cause
+minor changes. Backwards-compatibility will be maintained if reasonable.
+
+An API can be declared ``stable`` after fullfilling the following requirements:
+
+- Test cases for the new API with 100% coverage
+- Complete documentation in code. All public interfaces shall be documentated
+  and available in online documentation.
+- The API has been in-use and was available in at least 2 development releases
+
```

Comment:
Maybe add a comment saying that there can be additions to a stable API as long as they don't break compatibility?